napari/layers/labels/_labels_mouse_bindings.py

```diff
@@ -16,7 +37,13 @@ def paint(layer, event):
             last_cursor_coord, layer.coordinates, layer.brush_size
         )
         for c in interp_coord:
-            layer.paint(c, layer.selected_label, refresh=False)
+            if layer._mode == Mode.PAINT:
+                layer.paint(c, layer.selected_label, refresh=False)
+            elif (
+                layer._mode == Mode.FILL
+                and layer.get_value() != layer.selected_label
+            ):
+                layer.fill(c, layer.selected_label)
```

Comment:
We should add the `refresh=True` kwarg to `fill` and set it to be `False` when called inside this for-loop.

We should then add
```python
        if refresh is True:
            self._save_history()
```
and

```python
        if refresh is True:
            self.refresh()
```

to the top and bottom of the `fill` function, in the same way they are used in the `paint` function, you can copy the doc string over too.

Having this be false during the drag should be a slight performance improvement as we we'll only trigger refresh once per mouse move, not for every interpolated point, and it will better match how `paint` works

Reply:
@sofroniewn does that mean though that we won't see the result of the filling as we're doing it?

Reply:
No you'll see the result as you're doing it - every move of the mouse. This means you wont bombard with 10x the number of refreshes due to the "interpolation" function leading to many calls per mouse move due to the extra `for` loop

Reply:
Added! nice catch